UnityProject/Assets/Scripts/Util/EncryptionUtils.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Text;
+using System;
+using System.Security.Cryptography;
+using System.IO;
+
+namespace Util
+{
+	/// <summary>
+	/// General tool for encrypting and decrypting strings.
+	/// </summary>
+	public static class EncryptionUtils
+	{
+		public static string Encrypt(string clearText, string encryptionKey)
+		{
+			byte[] clearBytes = Encoding.Unicode.GetBytes(clearText);
+			byte[] salt = Encoding.ASCII.GetBytes("Cuben Pete");
+			using var encryptor = Aes.Create();
+			var pdb = new Rfc2898DeriveBytes(encryptionKey,  salt);
+			encryptor.Key = pdb.GetBytes(32);
+			encryptor.IV = pdb.GetBytes(16);
+			using var ms = new MemoryStream();
+			using (var cs = new CryptoStream(ms, encryptor.CreateEncryptor(), CryptoStreamMode.Write))
+			{
+				cs.Write(clearBytes, 0, clearBytes.Length);
+				cs.Close();
+			}
+			clearText = Convert.ToBase64String(ms.ToArray());
+
+			return clearText;
+		}
+
+
+		public static string Decrypt(string cipherText, string encryptionKey)
+		{
+			cipherText = cipherText.Replace(" ", "+");
+			byte[] cipherBytes = Convert.FromBase64String(cipherText);
+			byte[] salt = Encoding.ASCII.GetBytes("Cuben Pete");
+			using var encryptor = Aes.Create();
+			var pdb = new Rfc2898DeriveBytes(encryptionKey, salt);
+			encryptor.Key = pdb.GetBytes(32);
+			encryptor.IV = pdb.GetBytes(16);
+			using var ms = new MemoryStream();
+			using (var cs = new CryptoStream(ms, encryptor.CreateDecryptor(), CryptoStreamMode.Write))
+			{
+				cs.Write(cipherBytes, 0, cipherBytes.Length);
+				cs.Close();
+			}
+			cipherText = Encoding.Unicode.GetString(ms.ToArray());
+
+			return cipherText;
+		}
+
+		/// <summary>
+		/// Same as Decrypt but catches the exception in case the key isn't valid and returns an empty string instead.
+		/// </summary>
+		/// <param name="cipherText"></param>
+		/// <param name="encryptionKey"></param>
+		/// <returns></returns>
+		public static string DecryptSafely(string cipherText, string encryptionKey)
+		{
+			try
+			{
+				return Decrypt(cipherText, encryptionKey);
+			}
+			catch (CryptographicException)
+			{
+				return string.Empty;
+			}
+		}
+
+		/// <summary>
+		/// A fast but ugly encryption/decryption function.
+		/// </summary>
+		/// <param name="inputData">string to decrypt/encrypt</param>
+		/// <param name="key">EncryptionSecret</param>
+		/// <returns></returns>
+		public static string XOREncryptDecrypt(string inputData, int key)
```

Comment:
This is useless in it's current form. Why add this?
1) Requires int key
2) Encrypted text is not human  readable.

Reply:
> Encrypted text is not human readable.

that's... that's the point

Reply:
By non readable I mean it produces Unicode garbage that cannot be interpreted by human

Reply:
this has been fixed